### PR TITLE
Deprecate TypedMap symbols and replace with named functions

### LINKED ISSUE #10221: Multiple (non-implicit) parameters to a symbolic/infix should be avoided
In #10199 we added symbolic methods to TypeMap, but as signaled by @NthPortal we should avoid this. 

Triggered by: 

I'm a bit late to the party here, but multiple (non-implicit) parameters to a symbolic/infix method are highly discouraged,  and planned to be deprecated. See scala/scala-dev#496 and https://github.com/lampepfl/dotty/pull/4311#issuecomment-435593905.

_Originally posted by @NthPortal in https://github.com/playframework/playframework/pull/10199_

### PR DESCRIPTION
**Pull request by BillyAutrey**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #10221 

## Purpose

* Deprecates `-` and `+` Scala symbol methods in `TypedMap`. 
* Adds `updated` and `removed` functions that serve the same purpose.
* Changes outside calls to `+`, `-`, `$plus`, and `$minus` to `updated`/`removed`.
* Adds an ignore rule for MiMa compatibility - new signature doesn't break backwards compatibility.

In the future, postfix symbolic operators are going to be deprecated for methods with multiple parameters. Thus, this change is necessary to meet that need. The original ticket suggests removing _all_ of those operators.

## Background Context

I opted to deprecate the old methods, in case others are using them.

### CODE REVIEW

**Comment by BillyAutrey** [comment]
Next commit will update usages of `TypedMap.+()` and `TypedMap.-()` through the code base. Just want to get feedback on this approach first.

**Comment by mkurz** [comment]
So I had a look at #10199 again and the three `+` and `-` method were actually only added in that PR, which was never backported to 2.8.x. That means we would actually introduce already deprecated methods in Play 2.9 :wink: That does not make sense of course so I just removed them now (in the first commit I just pushed).

The same is true for the `addAttrs(varargs)` method of the Play Java API, I does not yet exist in Play 2.8.x. Since it does cause some problems like
* we have to workaround https://github.com/scala/bug/issues/11944
* and it is causing troubles for Scala 3 support (see #11550, you already commented there)

I think its best to just remove it for now to make our lifes easier. I don't think someone will miss it. Actually I was the one who introduced it for a project I was working on back then. However, it's "just" useful for the Java api, and we still keep the `addAttrs(java.util.List)` method, so it's still possible to pass an array via `addAttrs(Arrays.asList(...))`

I also updated the mima filters in the last PR.

Let's merge this and then we can rebase the Scala PR without the duplicate RequestImpl class on it.

**Comment by mkurz** [comment]
btw, I was using `git difftool 2.8.x -- core/play/src/main/java/play/libs/typedmap/TypedMap.java core/play/src/main/java/play/mvc/Http.java core/play/src/main/scala/play/core/j/JavaHelpers.scala` how the files from the 2.8.x branch differ from the main branch/your commit.

File: core/play/src/main/scala/play/core/j/JavaHelpers.scala [hunk 1]
```diff
@@ -263,16 +262,7 @@ class RequestHeaderImpl(header: RequestHeader) extends JRequestHeader {
...
 }
 
-/**
- * trait needed as workaround for https://github.com/scala/bug/issues/11944
- * Also see original pull request: https://github.com/playframework/playframework/pull/10199
- * sealed so that lack of implementation can't be accidentally used elsewhere
- */
-private[j] sealed trait RequestImplHelper extends JRequest {
-  override def addAttrs(entries: TypedEntry[_]*): JRequest = ???
-}
```
Comment (mkurz) [inline core/play/src/main/scala/play/core/j/JavaHelpers.scala:273]: Removing `addAttrs(entries: TypedEntry[_]*)´ allows us to get rid of this workaround for the mentioned Scala "bug" and make life easiser for #11550

**Review by mkurz — APPROVED** [review]
LGMT, thanks Billy!

**Pull request merged**